### PR TITLE
Screenshot directory configuration

### LINKED ISSUE #117: Add possibility to set custom screenshot directory
Currently, screenshots taken by WhiteLibrary are stored in the Robot Framework output folder.
But I would like to be able to change this location when loading the library or via a new keyword **`Set Screenshot Directory`**

When loading:
**`Library  WhiteLibrary  screenshot_directory=${OUTPUT_DIR}/Screens`**

Via keyword:
**`Set Screenshot Directory  ${OUTPUT_DIR}/Screens`**

This possibility is available in most other Robot framework libraries like the built-in `Screenshot` library and `SeleniumLibrary`.

### PR DESCRIPTION
**Pull request by eeter**
- New keyword `Set Screenshot Directory` 
- New library argument `screenshot_dir`

Closes #117

### CODE REVIEW

File: src/WhiteLibrary/__init__.py [hunk 1]
```diff
@@ -146,7 +151,7 @@ def __init__(self):
                           TreeKeywords(self),
                           UiItemKeywords(self),
                           WindowKeywords(self),
-                          ScreenshotKeywords(self)]
+                          ScreenshotKeywords(self, screenshot_dir)]
```
Comment (rasjani) [inline src/WhiteLibrary/__init__.py]: screenshotkeywords takes 2 parameters, state & screenshot_dir but only path is pased here ..

File: src/WhiteLibrary/keywords/screenshot.py [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
-
 from System.Drawing import Bitmap   # noqa: F401 #pylint: disable=unused-import
 from System.Drawing.Imaging import ImageFormat
 from TestStack.White import Desktop
 
 
 class ScreenshotKeywords(LibraryComponent):
-
-    def __init__(self, state):
+    def __init__(self, state, directory):
```
Comment (rasjani) [inline src/WhiteLibrary/keywords/screenshot.py]: see previous comment ..

File: src/WhiteLibrary/__init__.py (hunk 1 above)
Comment (eeter) [reply src/WhiteLibrary/__init__.py]: "self" (the instance of WhiteLibrary) is given as the state parameter, state is used similarly to ctx in SeleniumLibrary

File: src/WhiteLibrary/keywords/screenshot.py [hunk 3]
```diff
@@ -1,20 +1,34 @@
...
         super(ScreenshotKeywords, self).__init__(state)
         self.state.screenshooter = self
+        self._screenshot_directory = directory
+
+    @keyword
+    def set_screenshot_directory(self, path):
+        """Sets the directory where WhiteLibrary stores screenshots and returns the previously set path.
+
+        If the given directory does not exist in the system, it will be created.
```
Comment (Kompakti) [inline src/WhiteLibrary/keywords/screenshot.py]: Maybe point out that the directory will be created only when the first screenshot is taken?

**Comment by eeter** [comment]
Flake8 still gives error ".\src\WhiteLibrary\__init__.py:128:1: D403 First word of the first line should be properly capitalized"

**Comment by rasjani** [comment]
> Flake8 still gives error ".\src\WhiteLibrary__init__.py:128:1: D403 First word of the first line should be properly capitalized"

cuz the first comment - which pydocstyle considers as part of the files documentation is not capitalized ..  D403 could be just disabled if docstring formatting is not needed/considered worthy..

**Pull request merged**